web/pandas/pdeps/0015-do-not-require-pyarrow.md

```diff
@@ -0,0 +1,79 @@
+
+1) Requiring pyarrow as a dependency can cause installation problems for a significant portion of users.
+
+   - Pyarrow does not fit or has a hard time fitting in space-constrained environments
+such as AWS Lambda, due to its large size of around ~40 MB for a compiled wheel
+(which is larger than pandas' own wheel sizes)
+     - This can also cause problems for downstream libraries that use pandas as a dependency
+       as while pandas + pyarrow can potentially fit in an AWS Lambda environment, the combination of
+       pandas, pyarrow, and the downstream library may not fit.
+     - While it may potentially be possible to work around this issue by using the AWS Lambda Layer from
+       the [AWS SDK for pandas](https://aws-sdk-pandas.readthedocs.io/en/stable/install.html#aws-lambda-layer),
+       the primary benefit of pyarrow strings is not enough to force users to make a disruptive change.
+
+   - Installation of pyarrow is not possible on some platforms. We provide support for some
+less widely used platforms such as Alpine Linux, which pyarrow does not provide wheels for.
+     - While pyarrow has made great strides towards supporting most platforms that pandas is installable on
+       (e.g. the recent addition of pyodide support in pyarrow), we would still have to drop support for some
+       platforms like musllinux (the feature request is tracked [here](https://github.com/apache/arrow/issues/18036)) if pyarrow was to be required.
+
+   While installation issues are mentioned in the drawbacks section of PDEP-10, at the time of the writing
+of the PDEP, we underestimated the impact this would have on users, and also downstream developers.
+
+2) Many of the benefits presented in PDEP-10 can be materialized for users that have pyarrow installed, without
+   forcing a pyarrow requirement on other users.
+
+   In PDEP-10, there are three primary benefits listed:
+
+   - First class support for strings.
+
+     - PDEP-14 enables a new string dtype by default for pandas 3.0,
+       which will be backed by a pyarrow string dtype by default,
+       (for users who have pyarrow installed) and use a Python object based fallback for
+       users that don't have pyarrow installed. This allows all users to experience the usability
+       benefits of a string dtype by default, and for users with pyarrow to experience the performance
+       benefits of a pyarrow backed string array.
+
+   - Support for dtypes not present in pandas.
+     - There are some types in pyarrow that don't have a corresponding pandas/numpy dtype, for example
+       the nested pyarrow types(e.g. lists and structs), and decimal types.
+     - Currently, users can already create arrays with these dtypes if they have pyarrow installed, but we cannot infer
+     arrays to those dtypes by default, without forcing a pyarrow requirement on users,
+     as there is no Python/numpy equivalent for these dtypes).
+
+   - Interoperability
```

Comment:
I think @MarcoGorelli point was to remove the entire Interoperability section entirely, but if that's not true then I don't understand the point this is trying to make in its current form. 

The beneficiary of the Arrow C Data interface is not just other dataframe libraries - a decent listing can be found here:

https://github.com/apache/arrow/issues/39195#issuecomment-2245718008

For a direct benefit to pandas, it helps with I/O to boost performance, ensure proper data types, and reduce the amount of code burden. We have already seen this benefit with ADBC drivers, and from the link above, it looks like there is some near-term potential for it to help Excel I/O via fastexcel

Reply:
Sure, will take it out the next go.

Reply:
The fact that we don't require PyArrow might put is in a bind for downstream libraries that want interchange with pandas, but themselves probably aren't in a position to require PyArrow. In particular, this conversation is happening with seaborn:

https://github.com/mwaskom/seaborn/pull/3782#discussion_r1837038219

Somewhat unfortunately, this may mean that we are asked to put more maintenance work into the interchange protocol